### PR TITLE
feat(editor): resolve markdown images with note-relative or absolute paths

### PR DESCRIPTION
**Pull request by trapias**
## Problem

Markdown image embeds with note-relative paths (``, ``) or absolute paths (``) were not resolved by the renderer, leaving the embed broken. This is the default shape Obsidian and most markdown editors emit, so importing content from those tools left images visibly missing in Tolaria.

## Solution

Teach `vaultImages.ts` to resolve these forms in `resolveImageUrls`, and to emit note-relative paths back in `portableImageUrls` so saved markdown stays portable across machines. URLs with a scheme (`https:`, `data:`, …) are explicitly guarded so they keep flowing through to the browser unchanged.

The note path is threaded from the active tab through `editorBlockResolution`, `editorRawModeSync`, and `useEditorTabSwap` so resolution has the context it needs.

| URL form | Before | After |
|---|---|---|
| `attachments/x.png` | resolved | resolved (unchanged) |
| `![[x.png]]` (wikilink) | resolved | resolved (unchanged) |
| `./x.png`, `x.png` | broken | resolved against note directory |
| `../shared/x.png` | broken | resolved with `..` traversal |
| `/Users/me/x.png` | broken | resolved as-is |
| `https://…`, `data:…` | passed through | passed through (explicitly guarded) |

The image regex was also made non-greedy (`[^)"]+?` instead of `[^)\s"]+`) so URLs with raw spaces (``) match correctly when followed by an optional title.

## Files changed

- `src/utils/vaultImages.ts` — resolution + reverse-mapping helpers (`noteDirectoryPath`, `joinNoteRelativePath`, `relativeFromNoteDirectory`, `hasUrlScheme`, `isAbsolutePath`)
- `src/utils/vaultImages.test.ts` — +14 test cases
- `src/hooks/editorBlockResolution.ts` — thread `notePath` through `preProcessEditorMarkdown`
- `src/components/editorRawModeSync.ts` — thread the active tab path through `serializeEditorDocumentToMarkdown`
- `src/hooks/useEditorTabSwap.ts` — pass note path to `portableImageUrls` on editor changes

## Test plan

- [x] `pnpm vitest run src/utils/vaultImages.test.ts src/hooks/useEditorTabSwap.test.ts` — 89/89 pass
- [x] Full Vitest suite: 3742/3744 — the 2 failures (`blockNoteSuggestionWrapper.regression.test.tsx`, `blockNoteTableHandles.regression.test.ts`) are pre-existing on `eb6b58eb` and unrelated to this change (verified via stash-and-rerun)
- [x] `npx tsc --noEmit` passes
- [x] `pnpm lint` passes
- [ ] Manual: open a note with `` and a real file next to it → image renders
- [ ] Manual: open a note with `` → image renders, traversal honoured
- [ ] Manual: paste a clipboard image → still saved to `attachments/` and round-trips correctly
- [ ] Manual: a remote `` reference → still loads from network, untouched

## Notes

- Helps close a gap under the **Obsidian migration / coexistence** [Canny entry](https://tolaria.canny.io/feature-requests) (status: *planned*)
- Complementary to #521 (configurable media folders): that PR decides *where* new media is written; this PR decides *how* existing markdown images are read and round-tripped. Disjoint files.
- Pure frontend change — no Rust, no Tauri command surface changes, no settings.
- Happy to back out and post on Canny first if reviewers prefer that route per `CONTRIBUTING.md`.

### CODE REVIEW

File: src/utils/vaultImages.ts [hunk 1]
```diff
@@ -58,6 +71,62 @@ function withoutTrailingSlash(path: AbsolutePath): AbsolutePath {
   return path.replace(/\/+$/, '')
 }
 
+function noteDirectoryPath(notePath: NotePath): AbsolutePath {
+  const idx = Math.max(notePath.lastIndexOf('/'), notePath.lastIndexOf('\\'))
+  if (idx <= 0) return notePath
+  return notePath.slice(0, idx)
+}
```
[the remark this answers is not in the document]
Comment (trapias) [reply src/utils/vaultImages.ts:79]: Fixed in da5f8f48. `noteDirectoryPath` now splits the three real cases explicitly: `idx === -1` returns `'.'`, `idx === 0` returns the separator itself (`'/'` or `'\\'`), and `idx > 0` returns the prefix slice. Two regression tests added in `vaultImages.test.ts` covering both edge cases (note at filesystem root, note path without separator).

File: src/utils/vaultImages.ts [hunk 2]
```diff
@@ -106,29 +181,55 @@ function rewriteMarkdownImages(
...
-    return currentVaultAttachmentPath(url, vaultPath)
+    const attachmentPath = currentVaultAttachmentPath(url, vaultPath)
+    if (attachmentPath) return attachmentPath
+
+    if (notePath) {
+      return noteRelativeFromAssetUrl(url, notePath)
+    }
+
+    return null
   })
```
[the remark this answers is not in the document]
Comment (trapias) [reply src/utils/vaultImages.ts:239]: Fixed in da5f8f48. `portableImageUrls` now falls back to `decodeAssetPath(url)` when the asset URL cannot be mapped to `attachments/` or to a note-relative form, so the internal `asset://localhost` scheme never survives into saved markdown. One existing test updated to assert the new unwrap-to-absolute-path behavior, two new tests added including a round-trip check.

**Comment by trapias** [comment]
Both Codacy findings addressed in da5f8f48:

🔴 **HIGH RISK** — `noteDirectoryPath` edge cases at `idx === -1` and `idx === 0` (see review reply on `vaultImages.ts:79`).
🟡 **MEDIUM RISK** — `portableImageUrls` fallback to absolute filesystem path so `asset://` never persists in saved markdown (see review reply on `vaultImages.ts:239`).

Tests: `pnpm vitest run src/utils/vaultImages.test.ts` — 40/40 passing (2 new tests for HIGH RISK, 2 new tests + 1 updated for MEDIUM RISK).

**Comment by LucaRonin** [comment]
Merged after some tweaks — closing this now. Thanks again for the contribution!

**Comment by LucaRonin** [comment]
Merged after some tweaks — closing this now.
